File: src/common/web/data/entities/connector/ConnectorInstance.ts
import { AuthorizationState } from "../authorization/AuthorizationState";
import { type ConnectorID } from "./Connector";

/**
 * The connector instance ID type.
 */
export type ConnectorInstanceID = string;

/**
 * A configured connector instance (i.e., a connector the user has added to his configuration).
 *
 * @param instance_id - The ID of the connector instance.
 * @param connector_id - The assigned connector.
 * @param name - The name of this connector instance.
 * @param description - The instance description.
 * @param authorization_state: The authorization state.
 */
export class ConnectorInstance {
    public readonly instance_id: ConnectorInstanceID;

    public readonly connector_id: ConnectorID;

    public readonly name: string;
    public readonly description: string;

    public readonly authorization_state: AuthorizationState;

    public constructor(
        instanceID: ConnectorInstanceID,
        connectorID: ConnectorID,
        name: string,
        description: string = "",
        authState: AuthorizationState = AuthorizationState.NotAuthorized,
    ) {
        this.instance_id = instanceID;

        this.connector_id = connectorID;

        this.name = name;
        this.description = description;

        this.authorization_state = authState;
    }
}

File: src/common/py/data/entities/connector/connector_instance.py
import uuid
from dataclasses import dataclass, field

from dataclasses_json import dataclass_json

from .connector import ConnectorID
from ..authorization import AuthorizationState

ConnectorInstanceID = uuid.UUID


@dataclass_json
@dataclass(kw_only=True)
class ConnectorInstance:
    """
    A configured connector instance (i.e., a connector the user has added to his configuration).

    Attributes:
        instance_id: The ID of the connector instance.
        connector_id: The assigned connector.
        name: The name of this connector instance.
        description: The instance description.
        authorization_state: The authorization state.
    """

    instance_id: ConnectorInstanceID = field(default_factory=uuid.uuid4)

    connector_id: ConnectorID = ""

    name: str = ""
    description: str = ""

    authorization_state: AuthorizationState = AuthorizationState.NOT_AUTHORIZED
